plutus-core/testlib/PlutusCore/Generators/PIR/Common.hs

```diff
@@ -0,0 +1,74 @@
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE UndecidableInstances  #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module PlutusCore.Generators.PIR.Common where
+
+import Data.Bifunctor
+import Data.Map.Strict (Map)
+import Data.Map.Strict qualified as Map
+import GHC.Stack
+import Test.QuickCheck.Modifiers (NonNegative (..))
+import Test.QuickCheck.Property
+import Text.Pretty
+import Text.PrettyBy
+import Text.PrettyBy.Internal
+
+import PlutusCore.Default
+import PlutusCore.Name
+import PlutusCore.TypeCheck (defKindCheckConfig)
+import PlutusCore.TypeCheck.Internal (inferKindM, runTypeCheckM, withTyVar)
+import PlutusIR
+import PlutusIR.Error
+
+instance Testable (Either String ()) where
```

Comment:
should this go upstream?

Reply:
It baffles me that something like this isn't the default way of doing stuff with QuickCheck.

Reply:
:shrug:

Reply:
In any case, I don't think we can get it upstream. Particularly since `Either String ()` is kinda way too specific and

```haskell
instance Show e => Testable (Either e a)
```

would add quotes for `String`.

I just wanted to make it usable, I propose to leave such tiny unimportant things as is, so that we don't spend another month discussing them.